tests/testAstrometryTask.py

```diff
@@ -85,6 +85,47 @@ def testRadial(self):
         """
         self.doTest(afwGeom.RadialXYTransform([0, 1.01, 1e-7]))
 
+    def testUsedFlag(self):
+        """Test that the solver will record number of sources used to table
+           if it is passed a schema on initialization.
+        """
+        distortedWcs = afwImage.DistortedTanWcs(self.tanWcs, afwGeom.IdentityXYTransform())
+        self.exposure.setWcs(distortedWcs)
+        loadRes = self.refObjLoader.loadPixelBox(bbox=self.bbox, wcs=distortedWcs, filterName="r")
+        refCat = loadRes.refCat
+        refCentroidKey = afwTable.Point2DKey(refCat.schema["centroid"])
+        refFluxRKey = refCat.schema["r_flux"].asKey()
+
+        sourceSchema = afwTable.SourceTable.makeMinimalSchema()
+        measBase.SingleFrameMeasurementTask(schema=sourceSchema)  # expand the schema
+        config = AstrometryTask.ConfigClass()
+        config.wcsFitter.order = 2
+        config.wcsFitter.numRejIter = 0
+        #   schema must be passed to the solver task constructor
+        solver = AstrometryTask(config=config, refObjLoader=self.refObjLoader, schema=sourceSchema)
+        sourceCat = afwTable.SourceCatalog(sourceSchema)
+        sourceCentroidKey = afwTable.Point2DKey(sourceSchema["slot_Centroid"])
+        sourceFluxKey = sourceSchema["slot_ApFlux_flux"].asKey()
+        sourceFluxSigmaKey = sourceSchema["slot_ApFlux_fluxSigma"].asKey()
+
+        for refObj in refCat:
+            src = sourceCat.addNew()
+            src.set(sourceCentroidKey, refObj.get(refCentroidKey))
+            src.set(sourceFluxKey, refObj.get(refFluxRKey))
+            src.set(sourceFluxSigmaKey, refObj.get(refFluxRKey)/100)
+
+        results = solver.run(
+            sourceCat=sourceCat,
+            exposure=self.exposure,
+        )
+        #   check that the used flag is set the right number of times
+        count = 0
+        flagKey = sourceCat.getSchema().find("calib_astrometryUsed").key
+        for source in sourceCat:
```

Comment:
I think that I originally I tried this, or a slight variant, as I did in the photCal test.  But I got 'Record data is not contiguous in memory. ' error.   I did not stop to explore it, but just changed it to a loop.
Let me know if you think it matters.  Since I built this out of a test which already existed, I didn't particularly want to mess around with it.